src/reflect/scala/reflect/internal/JDK9Reflectors.java

```diff
@@ -0,0 +1,54 @@
+package scala.reflect.internal;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.jar.JarFile;
+
+public final class JDK9Reflectors {
```

Comment:
This pattern lets us compile on 8 but call APIs of 9 reflectively. The performance overhead of reflection is close to zero when calling through static final method handles.